sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/RowWriterFactory.java

```diff
@@ -135,10 +136,12 @@ BigQueryRowWriter<ElementT> createRowWriter(String tempFilePrefix, DestinationT
         throw new IllegalStateException(
             "createRowWriter called when schemaFactory is null; forgot to call prepare()?");
       }
+      @SuppressWarnings({
```

Comment:
Do not suppress the warning, but rather follow through the consequences of allowing null here.

Reply:
given that `schemaFactory` here is a `SerializableFunction`, I don't think that's really feasible.  Doing so would cause a large amount of checker failures elsewhere, and changing the interface would be a breaking API change.

Did you have something else in mind?

Reply:
`SerializableFunction` is parametric in input type and output type, it doesn't need to change. What I'm not clear on is whether it is safe to make `schemaFactory` have type `SerializableFunction<TableSchema, @Nullable Schema>`. I don't recall if that causes problems because there are places that do not properly check its return value.

Reply:
ah ok I see what you mean, I didn't realize you could annotate generic parameters like that as well.  Updated.

Reply:
Yea it is new in more recent versions of Java, and you have to use the checkerframework annotation because the one in the stdlib does not allow TYPE_USE occurrences. Thanks!